packages/contracts/contracts/voting/erc20/Erc20VotingManager.sol

```diff
@@ -0,0 +1,215 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.10;
+
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/utils/introspection/ERC165Checker.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/introspection/ERC165Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20VotesUpgradeable.sol";
+
+import "../../core/IDAO.sol";
+import "../../tokens/GovernanceERC20.sol";
+import "../../tokens/GovernanceWrappedERC20.sol";
+import "../../plugin/PluginManager.sol";
+import "./ERC20Voting.sol";
+
+contract Erc20VotingManager is PluginManager {
+    using Address for address;
+    using ERC165Checker for address;
+
+    /// @notice The logic contract of the `ERC20Voting`.
+    ERC20Voting private erc20VotingBase;
+
+    /// @notice The logic contract of the `GovernanceERC20`.
+    address public governanceERC20Base;
+
+    /// @notice The logic contract of the `GovernanceWrappedERC20`.
+    address public governanceWrappedERC20Base;
+
+    struct TokenConfig {
+        address addr;
+        string name;
+        string symbol;
+        GovernanceERC20.MintConfig mintConfig;
+    }
+
+    struct VoteConfig {
+        uint64 participationRequiredPct;
+        uint64 supportRequiredPct;
+        uint64 minDuration;
+    }
```

Comment:
On the SDK we are using "settings" instead of config.
- Plugin settings (hard limits)
  - `minTurnout`
  - `minSupport`
  - `minDuration`
- Proposal settings (chosen limits, per-proposal)